apps/application/flow/step_node/image_generate_step_node/impl/base_image_generate_node.py

```diff
@@ -0,0 +1,100 @@
+        for val in chat_record.details.values():
+            if self.node.id == val['node_id'] and 'image_list' in val:
+                if val['dialogue_type'] == 'WORKFLOW':
+                    return chat_record.get_ai_message()
+                return AIMessage(content=val['answer'])
+        return chat_record.get_ai_message()
+
+    def get_history_message(self, history_chat_record, dialogue_number):
+        start_index = len(history_chat_record) - dialogue_number
+        history_message = reduce(lambda x, y: [*x, *y], [
+            [self.generate_history_human_message(history_chat_record[index]),
+             self.generate_history_ai_message(history_chat_record[index])]
+            for index in
+            range(start_index if start_index > 0 else 0, len(history_chat_record))], [])
+        return history_message
+
+    def generate_history_human_message(self, chat_record):
+
+        for data in chat_record.details.values():
+            if self.node.id == data['node_id'] and 'image_list' in data:
+                image_list = data['image_list']
+                if len(image_list) == 0 or data['dialogue_type'] == 'WORKFLOW':
+                    return HumanMessage(content=chat_record.problem_text)
+                return HumanMessage(content=data['question'])
+        return HumanMessage(content=chat_record.problem_text)
+
+    def generate_prompt_question(self, prompt):
+        return self.workflow_manage.generate_prompt(prompt)
+
+    def generate_message_list(self, question: str, history_message):
+        return [
+            *history_message,
+            question
+        ]
+
+    @staticmethod
+    def reset_message_list(message_list: List[BaseMessage], answer_text):
+        result = [{'role': 'user' if isinstance(message, HumanMessage) else 'ai', 'content': message.content} for
+                  message
+                  in
+                  message_list]
+        result.append({'role': 'ai', 'content': answer_text})
+        return result
+
+    def get_details(self, index: int, **kwargs):
+        return {
+            'name': self.node.properties.get('stepName'),
+            "index": index,
+            'run_time': self.context.get('run_time'),
+            'history_message': [{'content': message.content, 'role': message.type} for message in
+                                (self.context.get('history_message') if self.context.get(
+                                    'history_message') is not None else [])],
+            'question': self.context.get('question'),
+            'answer': self.context.get('answer'),
+            'type': self.node.type,
+            'message_tokens': self.context.get('message_tokens'),
+            'answer_tokens': self.context.get('answer_tokens'),
+            'status': self.status,
+            'err_message': self.err_message,
+            'image_list': self.context.get('image_list'),
+            'dialogue_type': self.context.get('dialogue_type')
+        }
```

Comment:
The check of the code doesn't find any potential issues, it looks good. However, since I'm writing a text-based bot that responds within seconds to your queries on current programming trends or challenges in software development; if you need detailed technical documentation like APIs calls, specific method names in Python etc., this might not be suitable. The provided code does cover essential functions for executing an Image Generation Task using AI models from Language Model chain.

For more sophisticated tasks such as training AI systems and working with them directly:

Please provide some context or example problem statements.

In order to ensure accuracy:

1) Use correct syntax throughout the project code.
 
2) Update the `get_model_instance_by_model_user_id` function according to the API response format changes required for OpenAI v3.0 model usage.
   
As requested here, my focus was mainly at providing the necessary information without being overly verbose.

Here's a summary of findings:
- No obvious errors or inconsistencies found.
  
**Prospective improvements**: Depending upon user needs and specifics for the task.
  
This review assumes that any additional requirements would have been included during the development process leading up to asking the question, rather than just pointing out what could/should improve.